### ISSUE TITLE
Create RedirectFormHandler ADT to support testing.

### ISSUE THREAD

**Issue by andorp** [liqd/aula#339]
__Before:__

````haskell
redirectFormHandler
    :: (FormPage p, Page p, ActionM r m)
    => m p                       -- ^ Page representation
    -> (FormPagePayload p -> m (FormPageResult p)) -- ^ Processor for the form result
    -> ServerT (FormHandler p) m
````

__After:__
````haskell

data RedirectFormHandler m p =
    RedirectFormHandler (m p) (FormPagePayload p -> m (FormPageResult p))

redirectFormHandler
    :: (FormPage p, Page p, ActionM r m)
    => RedriectFormHandler m p
    -> ServerT (FormHandler p) m
````

It supports separation of concepts and the test suite can have direct access to the defined functions for a page handler, that prevents the divergence in the test implementation and application logic.

**Comment by andorp** [comment]
I am going to implement this change after the Acid-State changes get merged. I don't want to interfere with the that change at the moment.

**Comment by np** [comment]
I'm all for it but please provide field names and lenses for this type.

**Comment by andorp** [comment]
Roger that

**Comment by andorp** [comment]
Also `redirectFormHandler` will be called on the later steps.

**Comment by andorp** [comment]
Resolved in #371
